### PR TITLE
ASoC: SOF: amd: Fix for reading position updates from stream box & clocksource change

### PR DESCRIPTION
**Pull request by reddysujith**
This patches consists of fixes to read position updates from stream box and also configurable clock by clock source change.

### CODE REVIEW

**Review by bhiregoudar — DISMISSED** [review]

**Review by sunilkumardommati — DISMISSED** [review]

File: sound/soc/sof/amd/acp.c [hunk 1]
```diff
@@ -390,6 +390,7 @@ static int acp_power_on(struct snd_sof_dev *sdev)
 
 static int acp_reset(struct snd_sof_dev *sdev)
 {
+	const struct sof_amd_acp_desc *desc = get_chip_info(sdev->pdata);
```
Comment (ranj063) [inline sound/soc/sof/amd/acp.c:393]: please fix typos in the commit message "select" "selected as" , "internal clock to 96MHz" -> "Internal clock of...", "depends on usacase" -> "depending on usecase".

File: sound/soc/sof/amd/acp.c [hunk 2]
```diff
@@ -410,6 +411,7 @@ static int acp_reset(struct snd_sof_dev *sdev)
 	if (ret < 0)
 		dev_err(sdev->dev, "timeout in releasing reset\n");
 
+	snd_sof_dsp_write(sdev, ACP_DSP_BAR, desc->acp_clkmux_sel, 0x03);
```
Comment (ranj063) [inline sound/soc/sof/amd/acp.c]: @reddysujith it is not clear what you're writing here. Does 0x03 mean you are configuring the clock source to internal or external? And where is the default set?

File: sound/soc/sof/amd/acp.c (hunk 1 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp.c:393]: Done. Thank you

File: sound/soc/sof/amd/acp.c (hunk 2 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp.c]: Register has default value right (i.e internal clock source to 96Mhz(0x00)).  we changed from default to 0x03 ACLK(i.e external clock source).

**Review by plbossart — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Review by sunilkumardommati — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Pull request merged**